code/modules/food_and_drinks/pizzabox.dm

```diff
@@ -68,13 +68,30 @@
 			bomb_overlay.pixel_y = 5
 			add_overlay(bomb_overlay)
 	else
-		icon_state = "pizzabox[boxes.len + 1]"
+		icon_state = "pizzabox"
+		var/current_offset = 3
+		for(var/V in boxes)
+			var/obj/item/pizzabox/P = V
+			var/mutable_appearance/box_overlay = mutable_appearance(P.icon, P.icon_state)
+			box_overlay.pixel_y = current_offset
+			add_overlay(box_overlay)
+			current_offset += 3
 		var/obj/item/pizzabox/box = boxes.len ? boxes[boxes.len] : src
 		if(box.boxtag != "")
 			var/mutable_appearance/tag_overlay = mutable_appearance(icon, "pizzabox_tag")
 			tag_overlay.pixel_y = boxes.len * 3
 			add_overlay(tag_overlay)
 
+/obj/item/pizzabox/worn_overlays(isinhands, icon_file)
+	. = list()
+	var/current_offset = 2
+	if(isinhands && boxes.len)

```

Comment:
just before you ask, for each loops don't run if the list is empty, so the check is unnecessary.